[Exploratory View] Added step level filtering/breakdowns

shahzad31 opened the pull request:
## Summary

Fix https://github.com/elastic/uptime/issues/372

User can filter step level metrics using step selection.

User can also do breakdowns by step name

![image](https://user-images.githubusercontent.com/3505601/137489535-9f0bdfec-7cfd-4d4e-ad7b-338d80043a73.png)

elasticmachine commented:
Pinging @elastic/uptime (Team:uptime)

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/161396/)
* Commit: c9c26ce907fc4fc4b48eccd84369943dc9cd04d2
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [Check Types and Public API Docs](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/161396/execution/node/428/log)

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @shahzad31

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/161396/","number":"161396","commit":"c9c26ce907fc4fc4b48eccd84369943dc9cd04d2"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/161513/)
* Commit: 4154197f36670d750ee70dd9f535abc410f013b9
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-115182/4154197f36670d750ee70dd9f535abc410f013b9)
* This comment will update when the build is complete

### Failed CI Steps
* [Jest Unit Tests](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/161513/execution/node/190/log)

### History
* :broken_heart: [Build #161448](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/161448/) failed 4374a21d0d8a6f9fe396ad5f206c2fd5ad8bd986
* :broken_heart: [Build #161396](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/161396/) failed c9c26ce907fc4fc4b48eccd84369943dc9cd04d2

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @shahzad31

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/161396/","number":"161396","commit":"c9c26ce907fc4fc4b48eccd84369943dc9cd04d2"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/161448/","number":"161448","commit":"4374a21d0d8a6f9fe396ad5f206c2fd5ad8bd986"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/161513/","number":"161513","commit":"4154197f36670d750ee70dd9f535abc410f013b9"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/161531/)
* Commit: 92ce1e607e6a9c3356d2e00f38275105e9b5e079
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-115182/92ce1e607e6a9c3356d2e00f38275105e9b5e079)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/828074371ecdd1473624445779b8b5ca0c246e48) | [after](https://github.com/elastic/kibana/commit/92ce1e607e6a9c3356d2e00f38275105e9b5e079) | diff |
| --- | --- | --- | --- |
| `observability` | 363.1KB | 364.8KB | +1.7KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/828074371ecdd1473624445779b8b5ca0c246e48) | [after](https://github.com/elastic/kibana/commit/92ce1e607e6a9c3356d2e00f38275105e9b5e079) | diff |
| --- | --- | --- | --- |
| `observability` | 34.5KB | 34.9KB | +344.0B |

### History
* :broken_heart: [Build #161513](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/161513/) failed 4154197f36670d750ee70dd9f535abc410f013b9
* :broken_heart: [Build #161448](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/161448/) failed 4374a21d0d8a6f9fe396ad5f206c2fd5ad8bd986
* :broken_heart: [Build #161396](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/161396/) failed c9c26ce907fc4fc4b48eccd84369943dc9cd04d2

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @shahzad31

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/161396/","number":"161396","commit":"c9c26ce907fc4fc4b48eccd84369943dc9cd04d2"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/161448/","number":"161448","commit":"4374a21d0d8a6f9fe396ad5f206c2fd5ad8bd986"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/161513/","number":"161513","commit":"4154197f36670d750ee70dd9f535abc410f013b9"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/161531/","number":"161531","commit":"92ce1e607e6a9c3356d2e00f38275105e9b5e079"}]}
PIPELINE-->

shahzad31 commented:
> Can we hide the step dropdown if `ALL_VALUES` is selected.
> 
ALL_VALUES isn't even possible to select anymore in monitor name.

> I was unable to break down by step name, even when selecting a specific browser monitor. **Edit: this happened, even after I fixed my mappings, but I am unable to recreate now.** Also, we may need a popover to explain why breakdown by step name is disabled when unavailable. 

I have added the tool tooltip in case where it's not available.
> 
> We should consider moving fields onto new lines. It's getting crowded, and the flexible layout means that the inputs change size as the user is interacting with them.
> 
Have tried to handle this
> 
> When breaking down by step name, we see the step names for all monitors, not just the one selected. Yes this is a broad issue with filering and breakdown, i am planing to handle this by moving layer level filters to global query, i am having few discussions with lens team to handle this.

cc @dominiqueclarke

shahzad31 commented:
Hidden step name filter for non step level metrics like monitor duration/up pings/down pings

shahzad31 commented:
@elasticmachine merge upstream
